m4/slepc.m4

```diff
@@ -16,26 +16,47 @@ AC_DEFUN([CONFIGURE_SLEPC],
 
   AS_IF([test "$enableslepc" !=  no],
         [
-          dnl Test to see if SLEPC_DIR set by user.  If not set, then
-          dnl try to autodetect in a default directory
+          dnl If SLEPC_DIR is not set, we should try PETSC_DIR
+          dnl since SLPEc could live in the same place as PETSc
           AS_IF([test "x$SLEPC_DIR" = x],
-                [export SLEPC_DIR=/usr/lib/slepc])
+                [export SLEPC_DIR=$PETSC_DIR])
 
           dnl Test to see if SLEPC_DIR and petscversion were set by user or
           dnl autodetection.  If not set, then disable slepc, print a message.
           AS_IF([test "x$SLEPC_DIR" = "x" || test "x$petscversion" = "x"],
-                [
-                  enableslepc=no
-                  AC_MSG_RESULT(<<< SLEPc disabled.  Please set your "\$SLEPC_DIR" environment variable correctly to enable SLEPc. >>>)
-                ],
-                [
-                  AC_CHECK_HEADER([$SLEPC_DIR/include/slepcversion.h],
-                                  [SLEPC_INCLUDE="-I$SLEPC_DIR/include"],
+                 [
+                   dnl Test to see if SLEPC_DIR set by user.  If not set, then
+                   dnl try to autodetect in a default directory
+                   AC_CHECK_HEADER([/usr/lib/slepc/slepcversion.h],
+                                  [
+                                   export SLEPC_DIR=/usr/lib/slepc
+                                   SLEPC_INCLUDE="-I$SLEPC_DIR/include"
+                                  ],
                                   [
-                                    AC_MSG_RESULT(<<< Invalid "\$SLEPC_DIR" detected (slepcversion.h not found). SLEPc disabled. >>>)
-                                    unset SLEPC_DIR
                                     enableslepc=no
+                                    AC_MSG_RESULT(<<< SLEPc disabled.  Please set your "\$SLEPC_DIR" environment variable correctly to enable SLEPc. >>>)
                                   ])
+                ],
+                [
+                  AC_CHECK_HEADER([$SLEPC_DIR/include/slepcversion.h],
```

Comment:
I think I follow your logic here but it's a bit convoluted. It's also a bit unsatisfying to repeat the code that looks in `/usr/lib/slepc` for headers like this. I can probably take a crack at improving it if you are getting sick of this PR...

Reply:
As said before, I am not familiar with m4 so that I can not make the thing more efficient. I would appreciate if you could help.